### PR TITLE
mariadb: Enable slow query logging

### PR DESCRIPTION
**Pull request by rsalevsky**
Before the config cleanup some archs had slow_query_logging enabled.
Let's reenable it for all archs.

### CODE REVIEW

**Comment by jsuchome** [comment]
What is the meaning of slow query and why do we want it?

File: chef/cookbooks/mysql/recipes/server.rb [hunk 1]
```diff
@@ -66,6 +66,11 @@
 EOC
 end
 
+# FIXME: The configs installed by the package are currently not overwritable
```
Comment (jsuchome) [inline chef/cookbooks/mysql/recipes/server.rb]: So is it a packaging bug? Should we file a bug report so it gets fixed?

**Comment by nicolasbock** [comment]
@jsuchome The slow query log logs queries that are slow. You can tune what is considered slow with the `long_query_time` setting. The default is 10 seconds though (see https://mariadb.com/kb/en/the-mariadb-library/slow-query-log-overview/). Nice feature!

**Comment by rsalevsky** [comment]
@jsuchome Added the explanation to the commit message.

File: chef/cookbooks/mysql/recipes/server.rb (hunk 1 above)
Comment (rsalevsky) [inline chef/cookbooks/mysql/recipes/server.rb]: Done

File: chef/cookbooks/mysql/recipes/server.rb [hunk 2]
```diff
@@ -66,6 +66,11 @@
 EOC
 end
 
+# FIXME: The configs installed by the package are currently not overwritable (bsc#1055268)
+file "/etc/my.cnf.d/error_log.cnf" do
+  action :delete
```
Comment (rhafer) [inline chef/cookbooks/mysql/recipes/server.rb]: Hm, this means with every mariadb maintenance update we will get the file back, until chef-client runs again and deletes it. As chef-client won't restart the server after deleting the file (but rpm will do), we might endup running with the wrong configuration. I think I'd prefer to just ship a template for "/etc/my.cnf.d/error_log.cnf" for now. 

Or alternatively just add the "slow_query_log=1" settings to separate file "slow_query_log.cnf" and leave "/etc/my.cnf.d/error_log.cnf" in place. I think that should work as well.

**Comment by rsalevsky** [comment]
@rhafer I removed the part completely as the logging config gets loaded after the error_log config. So it's not a problem for us.

File: chef/cookbooks/mysql/recipes/server.rb [hunk 3]
```diff
@@ -78,6 +78,17 @@
...
 
+template "/etc/my.cnf.d/logging.cnf" do
+  source "logging.cnf"
+  owner "root"
+  group "mysql"
+  mode "0640"
+  variables(
+    slow_query_logging_enabled: node[:database][:mysql][:slow_query]
+  )
+  notifies :restart, "service[mysql]", :immediately
```
Comment (rhafer) [inline chef/cookbooks/mysql/recipes/server.rb]: Yet another immediate restart. This means in the worst case mysql will be restarted 3 times during a chef-client. But I am fine to merge this for now an address that later.

File: chef/cookbooks/mysql/recipes/server.rb [hunk 4]
```diff
@@ -78,6 +78,17 @@
   notifies :restart, "service[mysql]", :immediately
 end
 
+template "/etc/my.cnf.d/logging.cnf" do
+  source "logging.cnf"
```
Comment (jsuchome) [inline chef/cookbooks/mysql/recipes/server.rb]: `"logging.cnf.erb"`

**Pull request merged**